tests/codex/testvalidation.nim

```diff
@@ -67,7 +119,11 @@ asyncchecksuite "validation":
     check market.markedAsMissingProofs.len == 0
 
   test "it does not monitor more than the maximum number of slots":
+    let validationParams = initValidationParams(maxSlots, partitionSize = 1, partitionIndex = 0)
+    let validation = createValidation(clock, market, validationParams)
+    await validation.start()
     for _ in 0..<maxSlots + 1:
       let slot = Slot.example
       await market.fillSlot(slot.request.id, slot.slotIndex, proof, collateral)
+    await validation.stop()
     check validation.slots.len == maxSlots
```

Comment:
It should be covered to some extent by `slot is not observed if it is not in the partition` + other other tests, but I agree it is easier to create a couple of more documented cases by directly testing the functions you mention.

By the way, can you explain this:

![image](https://github.com/user-attachments/assets/fa911fb9-1ba0-46d1-8004-42a66fd721d3)

why does it complain about those two fields.

Also, are we measuring the coverage in the tests themselves?

Reply:
> can you explain this

Is this your editor? Is there a compiler error message?

> are we measuring the coverage in the tests themselves?

We are, but it's basic and we don't really use it because I don't think it's all that reliable. So yes, we have it as part of CI, but we don't really use it.

Reply:
> Is this your editor? Is there a compiler error message?

No, I just clicked the link somewhere the review (https://app.codecov.io/gh/codex-storage/nim-codex/pull/890/blob/codex/contracts/validation/validationparams.nim) and what I see does not make sense...

Or here: https://app.codecov.io/gh/codex-storage/nim-codex/pull/890

Also, I do not think it is good measuring coverage in the tests themselves... see this one: https://app.codecov.io/gh/codex-storage/nim-codex/pull/890/blob/tests/codex/testvalidation.nim

Reply:
Ah ok, yeah it's complaining that those lines aren't covered, which is obviously not correct since those field are used in the tests. As I said, it's not very reliable and no one is looking at the coverage stats for now until we can get a better coverage story for nim.

Reply:
Good to hear that. Coverage may be helpful but is also often source of false perception of quality. So I am not worried at all that we are not looking dogmatically at the coverage stats 👍.